build.gradle

```diff
@@ -220,6 +230,49 @@ testClusters.integTest {
             debugPort += 1
         }
     }
+
+    // This installs our plugin into the testClusters
+    plugin(project.tasks.bundlePlugin.archiveFile)
+    plugin(provider(new Callable<RegularFile>(){
+        @Override
+        RegularFile call() throws Exception {
+            return new RegularFile() {
+                @Override
+                File getAsFile() {
+                    return configurations.zipArchive.asFileTree.getSingleFile()
+                }
+            }
+        }
+    }))
+
+    // opensearch-geospatial plugin is being added to the list of plugins for the testCluster during build before
+    // the opensearch-job-scheduler plugin, which is causing build failures. From the stack trace, this looks like a bug.
+    //
+    // Exception in thread "main" java.lang.IllegalArgumentException: Missing plugin [opensearch-job-scheduler], dependency of [opensearch-geospatial]
+    //       at org.opensearch.plugins.PluginsService.addSortedBundle(PluginsService.java:515)
+    //
+    // A temporary hack is to reorder the plugins list after evaluation but prior to task execution when the plugins are installed.
+    // See https://github.com/opensearch-project/anomaly-detection/blob/fd547014fdde5114bbc9c8e49fe7aaa37eb6e793/build.gradle#L400-L422
+    nodes.each { node ->
+        def plugins = node.plugins
+        def firstPlugin = plugins.get(0)
+        plugins.remove(0)
+        plugins.add(firstPlugin)
+    }
+}
+
+testClusters.yamlRestTest {
```

Comment:
why yamlRestTest? is the RestIT are not sufficient?

Reply:
We already has yamlRestTest in geospatial package. This is not to break existing yamlRestTest.